redisless/src/server/tests/mod.rs

```diff
@@ -1,18 +1,73 @@
-use redis::{cmd, Commands, RedisResult};
+use redis::{cmd, Commands, Connection, RedisResult};
 use std::{thread::sleep, time::Duration};
 
 use crate::server::ServerState;
 use crate::storage::in_memory::InMemoryStorage;
 use crate::Server;
 
-#[test]
-#[serial]
-fn test_redis_implementation() {
-    let port = 3366;
+fn get_redis_client_connection(port: u16) -> (Server, Connection) {
     let server = Server::new(InMemoryStorage::new(), port);
     assert_eq!(server.start(), Some(ServerState::Started));
     let redis_client = redis::Client::open(format!("redis://127.0.0.1:{}/", port)).unwrap();
-    let mut con = redis_client.get_connection().unwrap();
+    (server, redis_client.get_connection().unwrap())
+}
+#[test]
+#[serial]
+fn test_incr_decr_commands() {
+    let (server, mut con) = get_redis_client_connection(3365);
+
+    let _: () = con.set("some_number", "12").unwrap();
+    let _ = con
+        .send_packed_command(
+            cmd("INCR")
+                .arg("some_number")
+                .get_packed_command()
+                .as_slice(),
+        )
+        .unwrap();
+    let some_number: u32 = con.get("some_number").unwrap();
+    assert_eq!(some_number, 13_u32);
+
+    let _: () = con.set("n", "100").unwrap();
+    let _ = con
+        .send_packed_command(cmd("DECR").arg("n").get_packed_command().as_slice())
+        .unwrap();
+    let n: u32 = con.get("n").unwrap();
+    assert_eq!(n, 99_u32);
+
+    let _: () = con.set("0", "12").unwrap();
+    let _ = con
+        .send_packed_command(
+            cmd("INCRBY")
+                .arg("0")
+                .arg("500")
+                .get_packed_command()
+                .as_slice(),
+        )
+        .unwrap();
```

Comment:
Instead of sending a custom command, u can call `con.incr(...)` since the redis-rs client does implement the [`INCRBY`](https://docs.rs/redis/0.20.1/redis/struct.Cmd.html#method.incr) command. You can do the same for any other command that is implemented directly by redis-rs, `DECRBY` is also there. :smile: